odo/backends/dask.py

```diff
@@ -22,6 +21,12 @@ def discover_dask_array(a, **kwargs):
     return from_numpy(a.shape, a.dtype)
 
 
+@discover.register(dd.Series)
+@discover.register(dd.DataFrame)
+def discover_dask_dataframe(df):
+    return var * discover(df.head()).measure
```

Comment:
Yeah, it requires a full graph computation (dataframes don't store their size as part of the metadata).